Robal/src/main/java/ac/project/Robal/controllers/AccountController.java

```
@@ -1,5 +1,766 @@
+	})
+	@PreAuthorize("hasAnyRole('ADMIN','CUSTOMER')")
+	@DeleteMapping("/customers/{id}")
+	public void deleteCustomer(Principal principal, @PathVariable Long id) throws NotFoundException {
+		logger.info("***deleteCustomer method accessed by " + principal.getName() + "***");
+		Account user = AccountUtil.getAccount(principal.getName());
+
+		if (user.getAccountId() == id
+				|| user.getRole() == Role.ADMIN) {
+			accountService.deleteCustomer(user.getEmail(), id);
+		}
+	}
+
+	// Delete Owner
+	@ApiOperation(value = "Delete an Owner", response = Owner.class)
+	@ApiResponses(value = {
+			@ApiResponse(code = 200, message = "Successfully deleted Owner account"),
+			@ApiResponse(code = 400, message = "Invalid input")
+	})
+	@PreAuthorize("hasAnyRole('ADMIN','OWNER')")
+	@DeleteMapping("/owners/{id}")
+	public void deleteOwner(Principal principal, @PathVariable Long id) throws NotFoundException {
+		logger.info("***deleteOwner method accessed " + " by " + principal.getName() + "***");
+		Account user = AccountUtil.getAccount(principal.getName());
+
+		if (user.getAccountId() == id
+				|| user.getRole() == Role.ADMIN) {
+			accountService.deleteOwner(user.getName(), id);
+		}
+	}
+
+	// Delete Admin
+	@ApiOperation(value = "Delete an Administrator", response = Owner.class)
+	@ApiResponses(value = {
+			@ApiResponse(code = 200, message = "Successfully deleted Administrator account"),
+			@ApiResponse(code = 400, message = "Invalid input")
+	})
+	@PreAuthorize("hasRole('ADMIN')")
+	@DeleteMapping("/admins/{id}")
+	public void deleteAdministrator(Principal principal, @PathVariable Long id) throws NotFoundException {
+		logger.info("***deleteAdministrator method accessed by " + principal.getName() + "***");
+		Account user = AccountUtil.getAccount(principal.getName());
+
+		if (user.getAccountId() == id
+				|| user.getRole() == Role.ADMIN) { // seems redundant here!
+			accountService.deleteAdministrator(user.getName(), id);
+		}

```

Seems like the delete methods should return an error, if the account is not authorized. Otherwise, they will think it is successful. But this is only because of the if statements workaround, so I'll deduct less (-0.5%)